### PR TITLE
Match truncated for the Gu/Obs bylines

### LINKED ISSUE #1325: Match photographers even if byline is truncated
e.g. 38e4b6756d0c13014a35f35f760a42042627658a has "Christopher Thomond for The Gua" due to the limit in length of 32 chars for the byline field. Other of his pictures have an untruncated longer by-line, so the truncation is probably due to an app rewriting the metadata before we get the file.

1. Can we get the full untruncated data from somewhere, as hinted by @paperboyo elsewhere? A quick exiftool dump of the metadata doesn't show the full version in any other field unfortunately.
2. Can we detect truncated for the GU bylines optimistically?

### PR DESCRIPTION
**Pull request by theefer**
Fixes #1325.

### CODE REVIEW

**Comment by akash1810** [comment]
:+1:

File: common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/ExtractGuardianCreditFromByline.scala [hunk 1]
```diff
@@ -5,12 +5,29 @@ import com.gu.mediaservice.model.ImageMetadata
...
+
+  val GuardianPrefix = Prefix("Guardian")
+  val ObserverPrefix = Prefix("Observer")
 
   override def clean(metadata: ImageMetadata): ImageMetadata = metadata.byline match {
     case Some(BylineForTheGuardian(byline, org)) => {
       val orgName = org.toLowerCase.capitalize
       metadata.copy(byline = Some(byline), credit = Some(s"The $orgName"))
     }
+    // Catch truncated bylines (EXIF allows 32 chars only)
```
Comment (paperboyo) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/ExtractGuardianCreditFromByline.scala]: Technically, that's IPTC's legacy `By-line` field. In EXIF it's called `Artist` and does not have said restrictions. Minor point, obv.

Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/ExtractGuardianCreditFromByline.scala]: Oops you're right, my typo.

Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/cleanup/ExtractGuardianCreditFromByline.scala]: Patched on master, thanks.

**Pull request merged**